### PR TITLE
Service Bus: Updating doc for receiveMessages API to include details of background auto lock renewal

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

**Review by lmolkova — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java [hunk 1]
```diff
@@ -504,6 +504,24 @@ IterableStream<ServiceBusReceivedMessage> peekMessages(int maxMessages, long seq
...
+     * run a background task that will continuously renew the lock before it expires. By default, the lock renew task will
+     * run for a duration of 5 minutes, this duration can be adjusted using
+     * the {@link ServiceBusReceiverClientBuilder#maxAutoLockRenewDuration(Duration)} API or can be turned off by
+     * setting it {@link Duration#ZERO}. A higher {@code maxMessages} value means an equivalent number of lock renewal
+     * tasks running in the client, which may put more stress on low CPU environments. Given each lock renewal is a network
+     * call to the broker, a high number of lock renewal tasks making multiple lock renew calls also may have an adverse
+     * effect in namespace throttling. Additionally, if certain lock renewal tasks fail to renew the lock on time because
+     * of low CPU, service throttling or overloaded network, then client may lose the lock on the messages, which will
+     * cause the application's attempts to settle (e.g., complete, abandon) those messages to fail. The broker may
+     * redeliver those messages, but if the settling attempts fail repeatedly beyond the max delivery count, then the message
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java:519]:
```suggestion
     * will redeliver those messages, but if the settling attempts fail repeatedly beyond the max delivery count, then the message
```

Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java:519]: Similar comment about one below.

**Review by conniey — APPROVED** [review]
lgtm

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java (hunk 1 above)
Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java:519]: updated as suggested.

**Pull request merged**